### DISCUSSION TITLE
does carla still needs gtk2 "for extended LV2 UIs support"?

### DISCUSSION THREAD

**Discussion by yuuyins** [falkTX/Carla#1747] [Q&A] [answered]
does carla still needs gtk2 "for extended LV2 UIs support"?

https://github.com/falkTX/Carla/blob/main/INSTALL.md#build-dependencies

**Comment by falkTX** [comment, marked as answer]
no, gtk2 support is possible through dlopen/dlsym stuff, gtk2 itself is not needed at build time

**Reply by yuuyins** [reply]
> possible through dlopen/dlsym stuff

did you mean that in some cases "dlopen/dlsym stuff" needs gtk2 at run time?

**Reply by falkTX** [reply]
yes, but if you are trying to use gtk2 based uis, you are supposed to have gtk2 libs on the system anyhow

**Reply by yuuyins** [reply]
@falkTX How common are dlopen/dlsym stuff? As it is optional, most users would not need/expect it? Would you personally add gtk2 as dependency to your Carla installation?

I'm asking because a Nixpkgs maintainer/reviewer wants me to remove gtk2 https://github.com/NixOS/nixpkgs/pull/217424#discussion_r1117655993 and even gtk3 https://github.com/NixOS/nixpkgs/pull/217424#discussion_r1119367242 as optional dependencies

**Reply by falkTX** [reply]
common? it is just system calls, they are always present. gtk2 and gtk3 can be removed completely, carla does not need them at build time and at runtime it will try to find gtk2 as needed when opening a gtk2 based lv2 ui.
for gtk4 this is no longer possible, but luckily we have no gtk4 based lv2 uis for now

**Reply by yuuyins** [reply]
i miscommunicate (i meant how common is that dlopen/dlsym may need gtk2 and gtk3 at runtime, how common users might face that situation). but this clears all my doubts, thank you!
